block/manager.go

```diff
@@ -170,16 +177,34 @@ func NewManager(
 		settlementClient: settlementClient,
 		retriever:        dalc.(da.BatchRetriever),
 		// channels are buffered to avoid blocking on input/output operations, buffer sizes are arbitrary
-		syncTargetDiode: diodes.NewOneToOne(1, nil),
-		syncCache:       make(map[uint64]*types.Block),
-		isSyncedCond:    *sync.NewCond(new(sync.Mutex)),
-		batchInProcess:  batchInProcess,
-		logger:          logger,
+		syncTargetDiode:       diodes.NewOneToOne(1, nil),
+		syncCache:             make(map[uint64]*types.Block),
+		isSyncedCond:          *sync.NewCond(new(sync.Mutex)),
+		batchInProcess:        batchInProcess,
+		shouldProduceBlocksCh: make(chan bool, 1),
+		logger:                logger,
 	}
 
 	return agg, nil
 }
 
+// Start starts the block manager.
+func (m *Manager) Start(ctx context.Context, isAggregator bool) error {
+	m.logger.Info("Starting the block manager")
+	if isAggregator {
+		m.logger.Info("Starting in aggregator mode")
+		// TODO(omritoptix): change to private methods
+		go m.ProduceBlockLoop(ctx)
+	}
+	// TODO(omritoptix): change to private methods
+	go m.RetriveLoop(ctx)
```

Comment:
the ones that can should be. 
I will add it to this PR (and it will actually also answer the previous comment as the registration will be needed to be prefixed with `go`)